astropy/io/fits/hdu/compressed/compressed.py

```diff
@@ -40,6 +40,8 @@
     DITHER_SEED_CLOCK,
 )
 
+__all__ = ["COMPRESSION_ENABLED", "CompImageHDU"]
```

Comment:
[According to PEP8 `__all__` should be placed after the module docstring, but before the imports](https://peps.python.org/pep-0008/#module-level-dunder-names), also in the other files.

Reply:
Except
For `__future__` imports, which precede everything but the module docstring.

Reply:
I tried to follow the convention in the rest of the library: 232 files define `__all__` after the imports (excluding `from __future__ ...` while only 22 define `__all__` before imports by my count. This is one of the PEP8 suggestions that no one know about or bother to follow if they do:
- none of the style checkers/formatters (pycodestyle, black, ruff, isort, etc.) support it, including extensions 
- `numpy`, `scipy`, `pandas`, and probably many more scientific libraries put `__all__` after imports

So I figured in this case it's better to be consistent than follow an unenforced guideline. But I'll defer to you all if you feel strongly about sticking to PEP8.

Reply:
Most of `astropy` places `__all__` below the imports and changing that is not a high priority, but I don't see a good reason why brand new `__all__` definitions should knowingly violate PEP8.

Reply:
We should open an issue on Ruff's isort to enable sorting `__all__` to be above the imports.

Reply:
[PEP8 tells us](https://peps.python.org/pep-0008/#a-foolish-consistency-is-the-hobgoblin-of-little-minds):

> A style guide is about consistency. Consistency with this style guide is important. Consistency within a project is more important. Consistency within one module or function is the most important.

All other files in `io` and `visualizations` and 94% of all astropy files (with `__all__`) place `__all__` after imports, so unless there's a explicit directive to change the style in all new code I think doing it just for this PR violates both consistency within one module and consistency within a project.

Reply:
https://github.com/astral-sh/ruff/issues/9324

Reply:
@eerovaher I suggest that since this discussion affects Astropy package-wide we should have it post getting this in.